### ISSUE TITLE
duplicated datasets

### ISSUE THREAD

**Issue by edoardopasolli** [waldronlab/curatedMetagenomicData#127]
some datasets appear to be duplicated in the latest release due to the change of "-" into "_" in the filename:

Castro-NallarE_2015.metaphlan_bugs_list.oralcavity
Castro_NallarE_2015.metaphlan_bugs_list.oralcavity

Heitz-BuschartA_2016.metaphlan_bugs_list.stool
Heitz_BuschartA_2016.metaphlan_bugs_list.stool

Obregon-TitoAJ_2015.metaphlan_bugs_list.stool
Obregon_TitoAJ_2015.metaphlan_bugs_list.stool

**Comment by schifferl** [comment]
I don't think they are duplicated but you are correct in thinking this is related to the change from `-` to `_`. Do you have a more concrete example I could look at?

**Comment by edoardopasolli** [comment]
e.g., "curatedMetagenomicData("*metaphlan_bugs*.oralcavity", dryrun = T)"

gives

"BritoIL_2016.metaphlan_bugs_list.oralcavity" 
"Castro-NallarE_2015.metaphlan_bugs_list.oralcavity" "Castro_NallarE_2015.metaphlan_bugs_list.oralcavity"
"HMP_2012.metaphlan_bugs_list.oralcavity"

**Comment by lwaldron** [comment]
I notice this too in devel, for three datasets (but not including HMP?):

```
> curatedMetagenomicData("*metaphlan*") %>% gsub("-", "_", .) %>% table() %>% sort() %>% tail(., 3)
Castro_NallarE_2015.metaphlan_bugs_list.oralcavity 
                                                 2 
    Heitz_BuschartA_2016.metaphlan_bugs_list.stool 
                                                 2 
     Obregon_TitoAJ_2015.metaphlan_bugs_list.stool 
                                                 2 
> 
```

**Comment by edoardopasolli** [comment]
Yes only these three datasets are affected by this issue:

Castro-NallarE_2015.metaphlan_bugs_list.oralcavity
Castro_NallarE_2015.metaphlan_bugs_list.oralcavity

Heitz-BuschartA_2016.metaphlan_bugs_list.stool
Heitz_BuschartA_2016.metaphlan_bugs_list.stool

Obregon-TitoAJ_2015.metaphlan_bugs_list.stool
Obregon_TitoAJ_2015.metaphlan_bugs_list.stool

**Comment by lwaldron** [comment]
I see that two tests were circumvented to allow this bug to enter release:

https://github.com/waldronlab/curatedMetagenomicData/commit/f4235cb25281feec35223509190d3c96f16e12f0
https://github.com/waldronlab/curatedMetagenomicData/commit/81f91b49fec946d72cc29db419373548d732f4a9

Now that there may be code out "in the wild" using both forms, we are supposed to go through the [18-month deprecation cycle](https://bioconductor.org/developers/how-to/deprecation/). This is complicated by the fact that these names can be either functions or a function argument to `curatedMetagenomicData()`. And because the package is being created by functions, so the deprecation process will require either additional functions or manual intervention with each rebuild.

The underscore `_` versions were only released on May 1, so the safest approach would be to initiate the deprecation process on those manually to have it done ASAP. Lucas, let us know your thoughts?

**Comment by lwaldron** [comment]
Just to clarify, I would like to deprecate the *underscore* versions and keep the hyphen versions, since the underscores are a munging of the original hyphenated author names and are only recently released. I will remove the underscores from the `curatedMetagenomicData()` function, @schifferl can you clarify what needs to be done in the generating pipeline?

**Comment by lwaldron** [comment]
The most recent build ( [changeset](https://github.com/cmirzayi/curatedMetagenomicData/commit/8b66dda809c23dbb2218afb241b1ddb333d8ad5e#diff-93971b06ab2d4f6d6cd3bc0be3ca10fe) ) contains the following munged names:

* Heitz_BuschartA_2016
* Castro_NallarE_2015
* Obregon_TitoAJ_2015

@schifferl can you advise on where the names get munged?

**Comment by cmirzayi** [comment]
@lwaldron I think the issue might be in `eset_name()`:

```
eset_name <- function(dataset_dir, data_type, bodysite) {
    basename(dataset_dir) %>%
    paste(., data_type, bodysite, sep = ".") %>%
    gsub("-", "_", .)
}
```
I'm assuming we want it to run on data_type and bodysite but not on dataset_dir. Perhaps this could be:

```
eset_name <- function(dataset_dir, data_type, bodysite) {
  basename(dataset_dir) %>%
    paste(., gsub("-", "_", data_type), gsub("-","_", bodysite), sep = ".") 
}
```

That seems to work for me:

```
> eset_name("./Castro-NallarE_2015", "marker-abundance", "left-leg")
[1] "Castro-NallarE_2015.marker_abundance.left_leg"
```

**Comment by lwaldron** [comment]
Nice patch, @cmirzayi. Would you rerun those three datasets like this? I
don't think we want to munge any part of the names, but as far as I recall
the data_type and bodysite as defined upstream contain no hyphens anyways.

Il mar 23 ott 2018, 12:47 PM cmirzayi <notifications@github.com> ha scritto:

**Comment by cmirzayi** [comment]
Yep I'll rerun them.

**Comment by lwaldron** [comment]
I've initiated the [deprecation process](https://bioconductor.org/developers/how-to/deprecation/), and added a unit test against the "-" -> "_" name munging. 

Immediately after release, the devel version should have the following `expect_warning` lines changed to `expect_error`:

https://github.com/waldronlab/curatedMetagenomicData/blob/9e54bcbb0f291c683bb9d80808ee4fb1ebca1382/tests/testthat/test-nomungednames.R#L6

And `.Deprecated()` should be changed to `.Defunct()` in two places:
1. https://github.com/waldronlab/curatedMetagenomicData/blob/9e54bcbb0f291c683bb9d80808ee4fb1ebca1382/R/zzz.R#L42
2. https://github.com/waldronlab/curatedMetagenomicData/blob/9e54bcbb0f291c683bb9d80808ee4fb1ebca1382/R/curatedMetagenomicData.R#L47

@cmirzayi can I assign that to you? The only thing I am still not sure about is whether the deprecated functions will still appear at all after Lori updates the metadata, which will not contain the munged dataset names in the current version. If not, we'll have to decide whether to go against Bioconductor guidelines and skip the deprecation process, or introduce a hack that makes these functions still work for the next cycle.

**Comment by lwaldron** [comment]
Rats, the new dataset `Bengtsson-PalmeJ_2015` still has its hyphen munged.

**Comment by lwaldron** [comment]
Duplicate and munged dataset names are resolved in Bioc 3.1 rc1 (https://github.com/waldronlab/curatedMetagenomicData/releases/tag/Bioc3.1_rc1)
